docs/tutorials/cloud_fit_deploy/cloud-aws-sagemaker-deployment.md

```diff
@@ -0,0 +1,216 @@
+# Deploying AutoGluon Models with AWS SageMaker
+:label:`cloud_aws_sagemaker_deploy`
+
+After learning how to train a model using AWS SageMaker :ref:`cloud_aws_sagemaker_fit`, in this section we will learn how to deploy 
+trained models using AWS SageMaker and Deep Learning Containers. 
+
+The full end-to-end example is available in [amazon-sagemaker-examples](https://github.com/aws/amazon-sagemaker-examples/tree/master/advanced_functionality/autogluon-tabular-containers) repository.
+
+## Pre-requisites
+Before starting ensure that the latest version of sagemaker python API is installed via (`pip install --upgrade sagemaker`). 
+This is required to ensure the information about newly released containers is available.
+
+## Endpoint Deployment - Inference Script
+
+To start using the containers, an inference script and the [wrapper classes](https://github.com/aws/amazon-sagemaker-examples/blob/master/advanced_functionality/autogluon-tabular-containers/ag_model.py) are required.
+When authoring an inference [scripts](https://github.com/aws/amazon-sagemaker-examples/blob/master/advanced_functionality/autogluon-tabular-containers/scripts/), 
+please refer to SageMaker [documentation](https://docs.aws.amazon.com/sagemaker/latest/dg/adapt-inference-container.html).
+
+Here is one of the possible inference scripts. 
+
+- the `model_fn` function is responsible for loading your model. It takes a `model_dir` argument that specifies where the model is stored. 
+
+- the `input_fn` function is responsible for deserializing your input data so that it can be passed to your model. It takes input data and 
+content type as parameters, and returns deserialized data. The SageMaker inference toolkit provides a default implementation that deserializes 
+the following content types: JSON, CSV, numpy array, NPZ.
+
+```{.python}
+from autogluon.tabular import TabularPredictor
+import os
+import json
+from io import StringIO
+import pandas as pd
+import numpy as np
+
+
+def model_fn(model_dir):
+    """loads model from previously saved artifact"""
+    model = TabularPredictor.load(model_dir)
+    globals()["column_names"] = model.feature_metadata_in.get_features()
+    return model
+
+
+def transform_fn(model, request_body, input_content_type, output_content_type="application/json"):
+
+    if input_content_type == "text/csv":
+        buf = StringIO(request_body)
+        data = pd.read_csv(buf, header=None)
+        num_cols = len(data.columns)
+
+        if num_cols != len(column_names):
+            raise Exception(
+                f"Invalid data format. Input data has {num_cols} while the model expects {len(column_names)}"
+            )
+
+        else:
+            data.columns = column_names
```

Comment:
This line would cause the input to lose the first row because the first row would first be loaded as a header and then got replaced. Noticed this during cloud predictor implementation ;)

Reply:
No, we don't read headers: `data = pd.read_csv(buf, header=None)`

Reply:
Oh yea. My mistake